code/go/0chain.net/smartcontract/storagesc/challenge.go

```diff
@@ -97,7 +97,7 @@ func (sc *StorageSmartContract) blobberReward(alloc *StorageAllocation, latestCo
 			zap.Int64("allocation expiry", int64(alloc.Expiration)),
 			zap.Int64("challenge time", int64(challengeCompletedTime)))
 
-		return nil
+		return fmt.Errorf("blobber reward failed: allocation is finalized")
```

Comment:
Same as I commented in another PR, see [here](https://github.com/0chain/0chain/pull/2511/files#r1225972543)